### PR TITLE
Changes necessary for running MSAL inside broker

### PR DESCRIPTION
**Pull request by oldalton**
* Correct broker response handling that validates calling app
* Return refresh token in MSIDTokenResult
* Return WPJ error to broker
* Redirect uri verifier

### CODE REVIEW

File: MSAL/src/util/ios/MSALRedirectUriVerifier.m [hunk 1]
```diff
@@ -26,63 +26,111 @@
...
 {
-    NSString *scheme = redirectUri.scheme;
-
-    if (![self verifySchemeIsRegistered:scheme error:error])
+#if AD_BROKER
+    // Allow the broker app to use a special redirect URI when acquiring tokens
+    if ([url isEqualToString:MSID_AUTHENTICATOR_REDIRECT_URI])
     {
-        return NO;
+        return YES;
```
Comment (unpluggedk) [inline MSAL/src/util/ios/MSALRedirectUriVerifier.m]: What does this indicate?
Should we return MSALRedirectUri created with authenticator_redirect_uri?

File: MSAL/src/util/ios/MSALRedirectUriVerifier.m [hunk 2]
```diff
@@ -26,63 +26,111 @@
...
-    NSString *redirectUriString = [NSString stringWithFormat:@"%@://%@", scheme, hostComponent];
-    return [NSURL URLWithString:redirectUriString];
+    NSString *redirectUri = [NSString stringWithFormat:@"msal%@://auth", clientId];
+    return [NSURL URLWithString:redirectUri];
+}
+
++ (NSURL *)defaultBrokerCapableRedirectUri
+{
+    NSString *bundleID = [[NSBundle mainBundle] bundleIdentifier];
+    NSString *redirectUri = [NSString stringWithFormat:@"msauth%@://auth", bundleID];
```
Comment (unpluggedk) [inline MSAL/src/util/ios/MSALRedirectUriVerifier.m]: nit: do we want to use the constant added in msid?

Comment (unpluggedk) [reply MSAL/src/util/ios/MSALRedirectUriVerifier.m]: and msauthv2, no?

File: MSAL/src/util/ios/MSALRedirectUriVerifier.m (hunk 1 above)
Comment (oldalton) [reply MSAL/src/util/ios/MSALRedirectUriVerifier.m]: Of course it should :) Fixed.

File: MSAL/src/util/ios/MSALRedirectUriVerifier.m (hunk 2 above)
Comment (oldalton) [reply MSAL/src/util/ios/MSALRedirectUriVerifier.m]: Constants added in msid are for the broker scheme. This is MSAL redirect uri prefix. I'd like to keep those 2 separate. 

I'm also not fully sure if it should have v2 or not, as broker scheme is something more internal and this URL will be shown to everyone as the default MSAL redirect uri. Thoughts?

**Pull request merged**